### PR TITLE
Committing project to new Development branch.

### PR DESCRIPTION
**Pull request by Micrified**
This PR should allow future branches to be merged without conflict.

### CODE REVIEW

File: SummerSchool/RGSAnnouncementDataModel.swift [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    
+    var id, title, description, author: String?
+    var date: Date?
+    
+    
+    /// MARK: - Protocol Methods.
+    
+    /// The model entity key.
+    static var entityKey: [String : String] = [
+        "entityName"    : "AnnouncementEntity",
```
Comment (mircealungu) [inline SummerSchool/RGSAnnouncementDataModel.swift:24]: all the values have camelCase starting with lowerCase, but AnnouceemntEntity is uppercase...

File: SummerSchool/RGSForumThreadTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    var author: String! {
+        didSet (oldAuthor) {
+            if (author != nil) {
+                authorLabel.text = author;
+            }
+        }
+    }
+    
+    // The title of the forum thread.
+    var title: String! {
```
Comment (mircealungu) [inline SummerSchool/RGSForumThreadTableViewCell.swift:44]: there should be somewhere a comment to explain this pattern of 
`didSet(oldStuff) { label.text = stuff }`
what is oldStuff?

File: SummerSchool/RGSForumViewController.swift [hunk 3]
```diff
@@ -8,14 +8,36 @@
 
 import UIKit
 
-class RGSForumViewController: RGSBaseViewController {
+class RGSForumViewController: RGSBaseViewController, UITableViewDelegate, UITableViewDataSource, UIScrollViewDelegate {
```
Comment (mircealungu) [inline SummerSchool/RGSForumViewController.swift]: i'm not exactly an expert in swift/iOS but does it make sense to have this class be both the view delegate and the data source?

**Review by mircealungu — COMMENTED** [review]
looks pretty good. 
a few questions / comments in there.

File: SummerSchool/RGSForumViewController.swift (hunk 3 above)
Comment (Micrified) [reply SummerSchool/RGSForumViewController.swift]: Delegate and data source methods are standard for using UITableViews. Being the DataSource simply means I have to implement methods that handle configuring the cells, and the delegate methods allow me to control table behavior.

File: SummerSchool/RGSForumThreadTableViewCell.swift (hunk 2 above)
Comment (Micrified) [reply SummerSchool/RGSForumThreadTableViewCell.swift:44]: Yes, I will add them. didSet (old<varname>) is called once a variable has changed. The old value  may be accessed and you can decide if you want to keep or discard it.

File: SummerSchool/RGSAnnouncementDataModel.swift (hunk 1 above)
Comment (Micrified) [reply SummerSchool/RGSAnnouncementDataModel.swift:24]: I can change this, but inside my database I have defined the entities without camel-case names, and they must match. I felt this was a worth exception. 
screen shot 2017-12-18 at 11 13 28 am

Comment (mircealungu) [reply SummerSchool/RGSAnnouncementDataModel.swift:24]: ah, it's ok then.

File: SummerSchool/RGSForumThreadTableViewCell.swift (hunk 2 above)
Comment (mircealungu) [reply SummerSchool/RGSForumThreadTableViewCell.swift:44]: In retrospect, it should have been obvious :)

File: SummerSchool/RGSForumViewController.swift (hunk 3 above)
Comment (mircealungu) [reply SummerSchool/RGSForumViewController.swift]: ok. you're the expert of this domain. though i should look into it more closely at some point :)

**Pull request merged**